### PR TITLE
Add support for tooltip offset option to be a function

### LINKED ISSUE #24223: Allow function in offset option for Tooltips and Popovers
Popper.js allow to pass function to this parameter it'll allow folks more flexibility on how they handle their offsets.
See : https://popper.js.org/popper-documentation.html#modifiers..offset.fn

Related to #24208
/CC @romanlex

### PR DESCRIPTION
**Pull request by jgolieb**
Fixes #24223

### CODE REVIEW

File: js/src/tooltip.js [hunk 1]
```diff
@@ -273,6 +273,16 @@ class Tooltip {
       const attachment = this._getAttachment(placement)
       this.addAttachmentClass(attachment)
 
+      const offsetConf = {}
+      if (typeof this.config.offset === 'function') {
+        offsetConf.fn = (data) => {
+          data.offsets = $.extend({}, data.offsets, this.config.offset(data.offsets, this.element) || {})
```
Comment (Johann-S) [inline js/src/tooltip.js]: please use the spread operator here

```js
data.offsets = {
  ...data.offsets,
  ...this.config.offset(data, this.element) || {}
}

**Review by Johann-S — CHANGES_REQUESTED** [review]
Can you move the logic in a private method please and can you add unit tests please

File: js/src/tooltip.js (hunk 1 above)
Comment (jgolieb) [reply js/src/tooltip.js]: Done.

**Comment by Johann-S** [comment]
@XhmikosR it seems we have some issues with Bundlesize, the github token is not found https://travis-ci.org/twbs/bootstrap/jobs/486082928#L833

File: js/tests/unit/tooltip.js [hunk 2]
```diff
@@ -1069,4 +1069,38 @@ $(function () {
...
+    assert.ok(typeof offset.offset === 'undefined')
+    assert.ok(typeof offset.fn === 'function')
+  })
+
+  QUnit.test('should create offset modifier correctly when offset option is not a function', function (assert) {
+    assert.expect(2)
+
+    var $trigger = $('<a href="#" rel="tooltip" data-trigger="click" title="Another tooltip"/>')
+      .appendTo('#qunit-fixture')
+      .bootstrapTooltip()
```
Comment (Johann-S) [inline js/tests/unit/tooltip.js]:
```js
var myOffset = 5
var $trigger = $('<a href="#" rel="tooltip" data-trigger="click" title="Another tooltip"/>')
  .appendTo('#qunit-fixture')
  .bootstrapTooltip({
    offset: myOffset,
  })
```

and after you'll be able to do:
```js
assert.strictEqual(offset.offset, myOffset)
```

I think it's more explicit like that

**Comment by Johann-S** [comment]
Just one comment and it'll be alright 👍

**Comment by XhmikosR** [comment]
@Johann-S the GH token isn't (and wasn't) available for PRs made by people without push rights, same as Browser Stack.

The build fails here due to the bundlesize limit. We'll need to increase the two bundle JS limits a bit.

**Review by Johann-S — APPROVED** [review]

**Pull request merged**